source/core/utilities.cc

```diff
@@ -94,7 +94,105 @@ make_table_tensors_scalars(
   return table;
 }
 
+void
+fill_table_from_file(TableHandler &    table,
+                     std::string       file,
+                     const std::string delimiter)
+{
+  std::string line;
+  table.clear();
+  std::ifstream myfile(file);
+  // open the file
+  if (myfile.is_open())
+    {
+      std::vector<std::string> vector_of_column_names;
+      std::vector<double>      line_of_data;
+
+      while (std::getline(myfile, line))
+        {
+          // read the line and clean the resulting vector
+          std::vector<std::string> list_of_words_base =
+            Utilities::split_string_list(line, delimiter);
+          std::vector<std::string> list_of_words_clean;
+          for (unsigned int i = 0; i < list_of_words_base.size(); ++i)
+            {
+              if (list_of_words_base[i] != "")
+                {
+                  list_of_words_clean.push_back(list_of_words_base[i]);
+                }
+            }
+          // check if the line is contained words or numbers
+          try
+            {
+              line_of_data = Utilities::string_to_double(list_of_words_clean);
+              for (unsigned int i = 0; i < line_of_data.size(); ++i)
+                {
+                  table.add_value(vector_of_column_names[i], line_of_data[i]);
+                }
+            }
+          catch (...)
+            {
+              // the line contains words we assume these are the column
+              vector_of_column_names = list_of_words_clean;
+            }
+        }
```

Comment:
This is a bit weird, why do it with this way where you test every line?
You know the first line contains the word, and then the rest of the lines contains the data. It would be a lot better to just read a first line, store the words, then read afterwards and store the data. The current way is worse, because for every single line you end up testing everything for no reason. This is also error prone.